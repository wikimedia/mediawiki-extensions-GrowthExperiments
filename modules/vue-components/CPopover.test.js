const { shallowMount } = require( '@vue/test-utils' );
const CPopover = require( './CPopover.vue' );

const renderComponent = ( props = {
	title: 'Some title',
	closeIcon: 'Close icon',
	closeIconLabel: 'Close label',
	headerIcon: 'Header icon',
	headerIconLabel: 'Header icon label'
}, slots = {
	trigger: '<button>Click me</button>',
	content: '<p>Some fancy content</p>'
} ) => shallowMount( CPopover, {
	props,
	slots
} );

describe( 'CPopover', () => {
	it( 'should be closed by default', () => {
		const wrapper = renderComponent();
		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'should display slot content when opened', () => {
		const wrapper = renderComponent();

		wrapper.vm.togglePopover();
		wrapper.vm.$nextTick().then( () => {
			expect( wrapper.element ).toMatchSnapshot();
			expect( wrapper.emitted() ).toHaveProperty( 'open' );
		} );
	} );

	it( 'should hide slot content when closed', () => {
		const wrapper = renderComponent();

		wrapper.vm.close();
		wrapper.vm.$nextTick().then( () => {
			expect( wrapper.element ).toMatchSnapshot();
			expect( wrapper.emitted() ).toHaveProperty( 'close' );
		} );
	} );

	it( 'should hide content when pressing on esc', () => {
		const wrapper = renderComponent();

		wrapper.vm.togglePopover();
		wrapper.vm.$nextTick()
			.then( () => wrapper.find( '.ext-growthExperiments-Popover__surface-container' ).trigger( 'keyup.esc' ) )
			.then( () => {
				expect( wrapper.element ).toMatchSnapshot();
				expect( wrapper.emitted() ).toHaveProperty( 'close' );
			} );
	} );
} );
