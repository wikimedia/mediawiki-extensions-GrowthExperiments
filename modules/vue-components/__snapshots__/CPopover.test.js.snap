// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CPopover should be closed by default 1`] = `
<div
  class="ext-growthExperiments-Popover"
>
  <div>
    <div>
      
      <button>
        Click me
      </button>
      
    </div>
    <div
      class="ext-growthExperiments-Popover__surface-container"
    >
      <!--v-if-->
    </div>
  </div>
</div>
`;

exports[`CPopover should display slot content when opened 1`] = `
<div
  class="ext-growthExperiments-Popover"
>
  <div>
    <div>
      
      <button>
        Click me
      </button>
      
    </div>
    <div
      class="ext-growthExperiments-Popover__surface-container"
    >
      <c-popper-stub
        icon="Close icon"
        icon-label="Close label"
        triggerref="[object HTMLDivElement]"
      />
    </div>
  </div>
</div>
`;

exports[`CPopover should hide content when pressing on esc 1`] = `
<div
  class="ext-growthExperiments-Popover"
>
  <div>
    <div>
      
      <button>
        Click me
      </button>
      
    </div>
    <div
      class="ext-growthExperiments-Popover__surface-container"
    >
      <!--v-if-->
    </div>
  </div>
</div>
`;

exports[`CPopover should hide slot content when closed 1`] = `
<div
  class="ext-growthExperiments-Popover"
>
  <div>
    <div>
      
      <button>
        Click me
      </button>
      
    </div>
    <div
      class="ext-growthExperiments-Popover__surface-container"
    >
      <!--v-if-->
    </div>
  </div>
</div>
`;
